### ISSUE TITLE
codacy-clang-tidy exceptions

### ISSUE THREAD

**Issue by tsteven4** [codacy/codacy-clang-tidy#20]
codacy-clang-tidy fails with my clang-tidy output.  It fails with 0.3.3 and dd7fd57197855deec982afde8bea21eb45b4d319.
I am running on Ubuntu bionic with clang tidy 6.
```
clang-tidy -version
LLVM (http://llvm.org/):
  LLVM version 6.0.0
  
  Optimized build.
  Default target: x86_64-pc-linux-gnu
  Host CPU: skylake
```
```
cat /etc/os-release 
NAME="Ubuntu"
VERSION="18.04.4 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.4 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```

There are two problems:
The first is

> Exception in thread "main" java.lang.IllegalArgumentException: 'other' is different type of Path
	at sun.nio.fs.UnixPath.relativize(UnixPath.java:429)
	at sun.nio.fs.UnixPath.relativize(UnixPath.java:43)
	at com.codacy.clangtidy.ClangTidyReportParser$.$anonfun$parse$1(ClangTidyReportParser.scala:16)
	at scala.collection.immutable.LazyList$.$anonfun$flatMapImpl$1(LazyList.scala:1015)
	at scala.collection.immutable.LazyList.scala$collection$immutable$LazyList$$state$lzycompute(LazyList.scala:218)
	at scala.collection.immutable.LazyList.scala$collection$immutable$LazyList$$state(LazyList.scala:217)
	at scala.collection.immutable.LazyList$.$anonfun$flatMapImpl$2(LazyList.scala:1026)
	at scala.collection.immutable.LazyList.scala$collection$immutable$LazyList$$state$lzycompute(LazyList.scala:218)
	at scala.collection.immutable.LazyList.scala$collection$immutable$LazyList$$state(LazyList.scala:217)
	at scala.collection.immutable.LazyList.isEmpty(LazyList.scala:228)
	at scala.collection.immutable.LazyList$LazyIterator.hasNext(LazyList.scala:1216)
	at scala.collection.IterableOps.groupBy(Iterable.scala:555)
	at scala.collection.IterableOps.groupBy$(Iterable.scala:552)
	at scala.collection.AbstractIterable.groupBy(Iterable.scala:921)
	at com.codacy.clangtidy.Converter.convert(Converter.scala:14)
	at com.codacy.clangtidy.Main$.main(Main.scala:14)
	at com.codacy.clangtidy.Main.main(Main.scala)

This can be fixed in ClangTidyReportParser.scala:

```
diff --git a/src/main/scala/com/codacy/clangtidy/ClangTidyReportParser.scala b/src/main/scala/com/codacy/clangtidy/ClangTidyReportParser.scala
index b184a95..981c89b 100644
--- a/src/main/scala/com/codacy/clangtidy/ClangTidyReportParser.scala
+++ b/src/main/scala/com/codacy/clangtidy/ClangTidyReportParser.scala
@@ -10,7 +10,7 @@ object ClangTidyReportParser {
   def parse(lines: Seq[String], relativizeTo: Path): Seq[ClangTidyResult] = {
     lines.flatMap {
       case result @ ResultRegex(pathStr, line, column, level, txt, checksList) =>
-        val path = relativizeTo.relativize(Paths.get(pathStr))
+        val path = relativizeTo.relativize(Paths.get(pathStr).toAbsolutePath())
         val firstCheck = checksList.split(",").headOption
 
         firstCheck
```

The second problem is 

> Exception in thread "main" java.nio.charset.MalformedInputException: Input length = 1
	at java.nio.charset.CoderResult.throwException(CoderResult.java:274)
	at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:339)
	at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
	at java.io.InputStreamReader.read(InputStreamReader.java:185)
	at java.io.BufferedReader.fill(BufferedReader.java:161)
	at java.io.BufferedReader.readLine(BufferedReader.java:326)
	at java.io.BufferedReader.readLine(BufferedReader.java:392)
	at scala.io.BufferedSource$BufferedLineIterator.hasNext(BufferedSource.scala:73)
	at scala.collection.immutable.LazyList$.scala$collection$immutable$LazyList$$stateFromIterator(LazyList.scala:1137)
	at scala.collection.immutable.LazyList$.$anonfun$stateFromIterator$1(LazyList.scala:1137)
	at scala.collection.immutable.LazyList.scala$collection$immutable$LazyList$$state$lzycompute(LazyList.scala:218)
	at scala.collection.immutable.LazyList.scala$collection$immutable$LazyList$$state(LazyList.scala:217)
	at scala.collection.immutable.LazyList.isEmpty(LazyList.scala:228)
	at scala.collection.immutable.LazyList$.$anonfun$flatMapImpl$1(LazyList.scala:1014)
	at scala.collection.immutable.LazyList.scala$collection$immutable$LazyList$$state$lzycompute(LazyList.scala:218)
	at scala.collection.immutable.LazyList.scala$collection$immutable$LazyList$$state(LazyList.scala:217)
	at scala.collection.immutable.LazyList$.$anonfun$flatMapImpl$2(LazyList.scala:1026)
	at scala.collection.immutable.LazyList.scala$collection$immutable$LazyList$$state$lzycompute(LazyList.scala:218)
	at scala.collection.immutable.LazyList.scala$collection$immutable$LazyList$$state(LazyList.scala:217)
	at scala.collection.immutable.LazyList.isEmpty(LazyList.scala:228)
	at scala.collection.immutable.LazyList$LazyIterator.hasNext(LazyList.scala:1216)
	at scala.collection.IterableOps.groupBy(Iterable.scala:555)
	at scala.collection.IterableOps.groupBy$(Iterable.scala:552)
	at scala.collection.AbstractIterable.groupBy(Iterable.scala:921)
	at com.codacy.clangtidy.Converter.convert(Converter.scala:14)
	at com.codacy.clangtidy.Main$.main(Main.scala:14)
	at com.codacy.clangtidy.Main.main(Main.scala)

This is due to the DEGREE SIGN ° characters in the clang tidy output on lines 35053 and 35091.

It seems clang tidy output is encoded in LATIN1 instead of UTF-8.  If I use iconv to change to UTF-8 then the second problem is avoided.

The standard output from clang-tidy is in the attached file:
[ct_standard_output.txt](https://github.com/codacy/codacy-clang-tidy/files/4641218/ct_standard_output.txt)

**Comment by tsteven4** [comment]
The first error also occurs with clang-tidy-9.  The second error does not, but this is because those two lines are not flagged by modernize-raw-string-literal with clang-tidy-9.

```
LLVM (http://llvm.org/):
  LLVM version 9.0.0
  
  Optimized build.
  Default target: x86_64-pc-linux-gnu
  Host CPU: skylake
```

**Comment by josemiguelmelo** [comment]
Hi @tsteven4

Thanks a lot for the contribution and for reporting both these problems, your suggestions were really useful 👍 
 
We opened a PR here https://github.com/codacy/codacy-clang-tidy/pull/21 to solve both the problems you mentioned. We added a command line argument that enables the user to specify the encoding (`--encoding`). This way, we give the possibility for the user to use a different encoding without the need to use `iconv` workaround.

These fixes were published on the new version (0.4.0).
